framework/src/utils/PetscSupport.C

```diff
@@ -407,7 +407,7 @@ petscNonlinearConverged(SNES snes,
   // Whether or not to force SNESSolve() take at least one iteration regardless of the initial
   // residual norm
   PetscBool force_iteration = PETSC_FALSE;
-#if !PETSC_VERSION_LESS_THAN(3, 8, 3) || !PETSC_RELEASE_LESS_THAN(3, 8, 3)
```

Comment:
Well the second one says `RELEASE`, what's the difference?

Reply:
PETSC_RELEASE_LESS_THAN return true, only when the release version of petsc is used and the version is less than 3.8.3.  And PETSC_VERSION_LESS_THAN does not care if it is a petsc release.